packages/query/src/actions.ts

```diff
@@ -203,15 +228,23 @@ export async function getTypes(
   const paramTypeOIDs = params.map((p) => p.oid);
   const returnTypesOIDs = fields.map((f) => f.typeOID);
   const usedTypesOIDs = paramTypeOIDs.concat(returnTypesOIDs);
-
   const typeRows = await runQuery(
-    `select oid, typname from pg_type where oid in (${usedTypesOIDs.join(
-      ',',
-    )})`,
+    `
+SELECT pt.oid, pt.typname, pt.typtype, pe.enumlabel
+FROM pg_type pt
+LEFT JOIN pg_enum pe ON pt.oid = pe.enumtypid
+WHERE pt.oid IN (${usedTypesOIDs.join(',')})`,
```

Comment:
Oh I was thinking about that...

Reply:
Okay so I couldn't resist. I generated the necessary SQL but I the issue i came up against was that I cannot run the generated query against `AsyncQueue`. It looks like the current interface needs the 'extended query' message to be supported on your wire package to support bindings as required by `IDatabaseConnection`. Perhaps we can handle binding substitution on this side of the pipe. It would be nice if `AsyncQueue` supported `IDatabaseConnection` other than that I think I have this self-hosted.

Reply:
Haha okay nope I think you do... I see it breaks down into the various parse messages. I will have to have a go at this in a future PR :)

I won't bother to ask why you implemented the wire protocol... it does seem pretty neat. But why did you :) ? Was there a particular frustration with pg, or a hard requirement, or was it just because it was there?

Reply:
Yes, self-hosting isn't ready yet :)
Back when I implemented the wire protocol, I couldn't find a good TS native package that provided packet level primitives for interacting with PostgreSQL. These low level messages are needed to query the DB for type data. Another reason was to potentially evolve PgTyped into a self-contained Postgres client. I think there is a good possibility to improve the reliability and performance of existing client, but this is a side goal and not a priority at the moment.